### PR TITLE
Detect and label CPU Model Name#270

### PR DESCRIPTION
**Pull request by AshwiniGhanwat**
**Changes made in the following files:**
- deployment/components/common/worker-mounts-combined.yaml
- deployment/components/common/worker-mounts.yaml
- deployment/helm/node-feature-discovery/templates/worker.yaml
- pkg/nfd-client/worker/nfd-worker.go
- source/config.go
- source/source_test.go
- test/e2e/e2e-test-config.example.yaml
- test/e2e/node_feature_discovery.go

**New file to generate the detect CPU Info:**
- source/cpuinfo/cpuinfo.go

**Please find below tested output:**
image

### CODE REVIEW

**Comment by digambar15** [comment]
Request Maintainer to add /ok-to-test tag to initiate CI tests

**Comment by zvonkok** [comment]
@digambar15 Can you run `make gofmt-verify` before checking in?

File: test/e2e/e2e-test-config.example.yaml [hunk 1]
```diff
@@ -77,12 +82,22 @@ defaultFeatures:
         "feature.node.kubernetes.io/cpu-cpuid.MPX": "true"
         "feature.node.kubernetes.io/cpu-cpuid.RTM": "true"
         "feature.node.kubernetes.io/cpu-hardware_multithreading": "true"
+        "feature.node.kubernetes.io/cpuinfo-cpuinfo.model": "142"
+        "feature.node.kubernetes.io/cpuinfo-cpuinfo.cpu_MHz": "1896.002"
+        "feature.node.kubernetes.io/cpuinfo-cpuinfo.cpu_family": "6"
+        "feature.node.kubernetes.io/cpuinfo-cpuinfo.vendor_id": "GenuineIntel"
+        "feature.node.kubernetes.io/cpuinfo-cpuinfo.stepping": "10"
```
Comment (ArangoGutierrez) [inline test/e2e/e2e-test-config.example.yaml:89]: This test could fail if not run on the same cpu-arch

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]
badly indented
Also a comment about the test file

File: source/cpuinfo/cpuinfo.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2018-2021 The Kubernetes Authors.
```
Comment (ArangoGutierrez) [inline source/cpuinfo/cpuinfo.go:2]: If this is a new file, only 2021

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]

File: source/cpuinfo/cpuinfo.go (hunk 2 above)
Comment (ArangoGutierrez) [reply source/cpuinfo/cpuinfo.go:2]: PING*

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]

**Review by marquiz — CHANGES_REQUESTED** [review]
Thanks for the PR @AshwiniGhanwat 

I think this should be implemented as part of the `source/cpu` instead of a completely new source. Even the label name suggests that: `cpu-cpuinfo.model` makes more sense (IMO) than `cpuinfo-cpuinfo.model`.

Also, I'm not entirely sure how useful `cpu_MHz` label really is. Any thoughts?

**Comment by ArangoGutierrez** [comment]
> Thanks for the PR @AshwiniGhanwat
> 
> I think this should be implemented as part of the `source/cpu` instead of a completely new source. Even the label name suggests that: `cpu-cpuinfo.model` makes more sense (IMO) than `cpuinfo-cpuinfo.model`.
> 
> Also, I'm not entirely sure how useful `cpu_MHz` label really is. Any thoughts?

*PING* @AshwiniGhanwat

**Comment by AshwiniGhanwat** [comment]
Closing this PR. Included all comments in https://github.com/kubernetes-sigs/node-feature-discovery/pull/700

**Pull request closed without merging**